### PR TITLE
Migrated to MUI v5 - Azure DevOps, DevTools, and Linguist

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

Migrated the Azure DevOps, DevTools, and Linguist frontend plugins from Material UI v4 to v5

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by awanlin** [comment]
Hi @Rugvip and @tudi2d, took a swing at these, thought they would be low hanging fruit and good example but I've run into some odd issues mostly around `makeStyles` and using `theme` with it. After running `yarn tsc:full` I'm getting "Namespace 'React' has no exported member 'JSX'" errors that I can't seem to figure out 🤔

**Review by tudi2d — CHANGES_REQUESTED** [review]
Thanks for this PR - great work & much appreciated!! A minor thing is that you added the generic `Theme` types for `makeStyles`. Those shouldn't be required as we override the original types now. Still I ran into a similar issue so it would be interesting to find out how to fix linting locally. 

For the `Namespace 'React' has no exported member 'JSX'.` issue I would assume that this is related to the requirement for `"react"` dependencies from MUI to be `^17.0.0` ([Doc](https://mui.com/material-ui/getting-started/supported-platforms/#react))

At the moment it's the following for all 3 plugins:
```
"peerDependencies": {
    "react": "^16.13.1 || ^17.0.0",
    "react-dom": "^16.13.1 || ^17.0.0",
```

**Comment by tudi2d** [comment]
> For interested readers, here's the React 18 migration issue to track: #12252

**Comment by tudi2d** [comment]
> For interested readers, here's the React 18 migration issue to track: #12252

Thanks - just want to make clear it "only" has to be `^17.0.0`

**Comment by freben** [comment]
Yep but our versioning strategy for these framework libraries is generally to support the last two majors.

File: plugins/linguist/src/components/LinguistCard/LinguistCard.tsx [hunk 1]
```diff
@@ -14,24 +14,22 @@
...
+import { useTheme, Theme } from '@mui/material/styles';
+import makeStyles from '@mui/styles/makeStyles';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Tooltip from '@mui/material/Tooltip';
+import Chip from '@mui/material/Chip';
 
-const useStyles = makeStyles(theme => ({
+const useStyles = makeStyles<Theme>(theme => ({
```
[the remark this answers is not in the document]
Comment (awanlin) [reply plugins/linguist/src/components/LinguistCard/LinguistCard.tsx:32]: Honestly this was just me trying to work around type errors. I was using your GraphiQL PR as a guide but in some cases this seems to fixed things. I'm happy to pull it out as it matches better what you have in your PR and what the MUI v5 guide says

**Comment by awanlin** [comment]
@freben @tudi2d I'm going to try using just React 17 to confirm if that's causing the issues and if it is then put this on hold and I'll help out with the React migration more

File: plugins/linguist/src/components/LinguistCard/LinguistCard.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (awanlin) [reply plugins/linguist/src/components/LinguistCard/LinguistCard.tsx]: Alright, when I updated my branch to use React 17 all my errors went away so I've removed all the places where I was using the Theme type.

**Comment by awanlin** [comment]
@tudi2d @freben @Rugvip going to put this as Draft for now as I can't move forward until the React 18 migration is done

**Comment by awanlin** [comment]
I'm going to close this one seeing as it can't move forward without the React upgrade being completed and with the summer vacations that's going to be slow going. Will submit a fresh PR when the time is right 👍

**Pull request closed without merging**